tabular/src/autogluon/tabular/models/abstract/abstract_model.py

```diff
@@ -707,3 +701,70 @@ def save_info(self) -> dict:
         json_path = self.path + self.model_info_json_name
         save_json.save(path=json_path, obj=info)
         return info
+
+    def _get_types_of_features(self, df, skew_threshold=None, embed_min_categories=None, use_ngram_features=None, needs_torch=False, needs_extra_types=True):
```

Comment:
Could you make an AbstractNeuralNetworkModel which inherits from AbstractModel that just contains this method? I don't think this method should live in AbstractModel as it isn't generic enough. It may also make sense to clean this up so add a '# TODO: v0.1 clean method'

Reply:
Made an additional class wrapper and made TT and tabular_nn inherit from it.